compair/static/modules/answer/answer-module.js

```diff
@@ -211,11 +209,22 @@ module.controller(
             );
         };
 
-        $scope.trackExited = function() {
+        // detect when modal has made changes that require the page to be refreshed to show new information
+        $scope.refreshOnExit = false;
+        $scope.$on('modal.closing', function(event, reason, closed) {
+            if ($scope.refreshOnExit) {
```

Comment:
you can move this into both of the `$scope.modalInstance.result.then` functions below